pkg/api/http/actors.go

```diff
@@ -193,13 +193,13 @@ func (a *api) onCreateActorReminder(w http.ResponseWriter, r *http.Request) {
 	err = a.universal.Actors().CreateReminder(ctx, &req)
 	if err != nil {
 		if errors.Is(err, actors.ErrReminderOpActorNotHosted) {
-			msg := messages.ErrActorReminderOpActorNotHosted
+			msg := messages.ErrActorReminderOpActorNotHosted.RecordAndGet()
```

Comment:
Changes related to above: https://github.com/dapr/dapr/pull/8256/commits/7c6a69175d553c82f11b99f847ca5108000e3a9b

I've went through and pushed the metric recording up in the call stack as far as I could understand, and ended up with a few functions that take different structures depending on how the end of the chain's parameters were / how the error's structure ended up looking like:

1. `RecordAPIErrorCode()`
    * Used at the end of most HTTP responses, which are provided an APIError struct
2. `RecordErrorCode()`
    * Used at the end of various chains, which are provided the new ErrorCode struct, like some other HTTP responses, the HTTP error response helper `NewErrorResponse()`, some crypto gRPC errors, Pubsub error builder
3. `RecordCompAndGet()`
    * Used at the end of various chains, which are provided various composited errors not following the `"ERR_X_X_X" format, some general error builders, scheduler errors, state error builder


In my opinion, it might be somewhat less understandable than having the recording added to each usage of the error codes, but it did lower the lines changed drastically and seems like a workable solution for the time being.